operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/event/ReconciliationDispatcher.java

```diff
@@ -38,11 +38,17 @@
 
   private final Controller<P> controller;
   private final CustomResourceFacade<P> customResourceFacade;
+  // this is tp handle corner case, when there is a retry, but it is actually limited to 0.
+  // Usually for testing purposes.
+  private final boolean retryConfigurationHasZeroAttempts;
 
   ReconciliationDispatcher(Controller<P> controller,
       CustomResourceFacade<P> customResourceFacade) {
     this.controller = controller;
     this.customResourceFacade = customResourceFacade;
+
+    var retry = controller.getConfiguration().getRetry();
+    retryConfigurationHasZeroAttempts = retry == null || retry.initExecution().isLastAttempt();
```

Comment:
Great. Didn't think of calling it in the ctor. I was convince that it would only work in a reconciliation context. LGTM